frontend/src/api/user.ts

```diff
@@ -0,0 +1,100 @@
+import { AxiosResponse } from "axios";
+import { SolvedQuestion } from "../models/SolvedQuestion.model"; // Import the SolvedQuestion model
+import { userServiceClient } from "./server";
+
+/**
+ * Fetch questions completed by the user.
+ * @param userId - The ID of the user for whom you want to fetch completed questions.
+ * @returns An array of completed questions.
+ */
+export async function fetchUserCompletedQuestions(
+  userId: number
+): Promise<SolvedQuestion[]> { 
+  try {
+    const response: AxiosResponse = await userServiceClient.get(
+      `/api/users/${userId}/questions`
+    );
+
+    const resData = response.data;
+    const completedQuestions: SolvedQuestion[] = resData.map(
+      (q: any) =>
+        new SolvedQuestion(
+          q._id,
+          q.id,
+          q.title,
+          q.description,
+          q.topics,
+          q.difficulty,
+          false, // Default value for solved
+          undefined // Default value for solvedDate
+        )
+    );
+
+    return completedQuestions;
+  } catch (error) {
+    // Handle errors appropriately, e.g., log the error, return an empty array, or throw an error.
+    console.error(error);
+    throw error;
+  }
+}
+
+/**
+ * Add a new question to the user's completed questions.
+ * @param userId - The ID of the user to whom you want to add the question.
+ * @param questionId - The ID of the question you want to add.
+ * @param complexity - The complexity of the question.
+ * @param category - The category of the question.
+ * @returns The added question.
+ */
+export async function addUserQuestion(
+  userId: number,
+  questionId: string,
+  complexity: number,
+  category: string[]
+): Promise<SolvedQuestion> { 
+  try {
+    const response: AxiosResponse = await userServiceClient.post(
+      `/api/users/${userId}/addquestions`,
+      {
+        userId,
+        questionId,
+        complexity,
+        category,
+      }
+    );
+
+    const resData = response.data;
+    const addedQuestion: SolvedQuestion = new SolvedQuestion(
+      resData._id,
+      resData.id,
+      resData.title,
+      resData.description,
+      resData.topics,
+      resData.difficulty,
+      false, // Default value for solved
+      undefined // Default value for solvedDate
+    );
+
+    return addedQuestion;
+  } catch (error) {
+    // Handle errors appropriately, e.g., log the error or throw it to be handled by the caller.
+    console.error(error);
+    throw error;
+  }
+}
+
+
+/**
+ * Fetch the user's ID.
+ * @returns The user's ID or null if not found.
+ */
+export async function fetchUserId(): Promise<number | null> {
+  try {
+    const response: AxiosResponse = await userServiceClient.get("/api/get-user-id");
+    return response.data.userId;
+  } catch (error) {
+    // Handle errors appropriately, e.g., log the error or return null.
+    console.error("Error fetching user ID:", error);
+    return null;
+  }
+}
```

Comment:
dont think `/api/get-user-id` is a valid endpoint on the backend.

besides, why do we need to have an endpoint for fetching the current user's id? When the user is logged in, the user's information is stored in the `authSlice`. We can access the user info by using `selectUser`.